wandb/sdk/wandb_artifacts.py

```diff
@@ -1154,9 +1143,8 @@ def __init__(self, scheme: Optional[str] = None) -> None:
         """
         self._scheme = scheme or ""
 
-    @property
-    def scheme(self) -> str:
-        return self._scheme
+    def can_handle(self, parsed_url: "ParseResult") -> bool:
+        return parsed_url.scheme == self._scheme
```

Comment:
I have a much larger soapbox on how we overuse inheritance, but since we already are here, why not just make this the default implementation and only override it for the single class that needs it (your new one)? Otherwise this seems pretty repetitive.

Reply:
Right, I'd rather not put logic that only applies to some subclasses in the parent class. Btw, this code isn't longer than the previous one.